api/src/repositories/survey-repository.ts

```diff
@@ -345,6 +374,59 @@ export class SurveyRepository extends BaseRepository {
     return result;
   }
 
+  /**
+   * SQL query to get the list of messages for an occurrence submission.
+   *
+   * @param {number} submissionId
+   * @returns {SQLStatement} sql query object
+   */
+  async getOccurrenceSubmissionMessages(submissionId: number): Promise<IOccurrenceSubmissionMessagesResponse[]> {
+    const sqlStatement = SQL`
+      SELECT
+        sm.submission_message_id as id,
+        smt.name as type,
+        sst.name as status,
+        smc.name as class,
+        sm.message
+      FROM
+        occurrence_submission as os
+      LEFT OUTER JOIN
+        submission_status as ss
+      ON
+        os.occurrence_submission_id = ss.occurrence_submission_id
+      LEFT OUTER JOIN
+        submission_status_type as sst
+      ON
+        sst.submission_status_type_id = ss.submission_status_type_id
+      LEFT OUTER JOIN
+        submission_message as sm
+      ON
+        sm.submission_status_id = ss.submission_status_id
+      LEFT OUTER JOIN
+        submission_message_type as smt
+      ON
+        smt.submission_message_type_id = sm.submission_message_type_id
+      LEFT OUTER JOIN
+        submission_message_class smc
+      ON
+        smc.submission_message_class_id = smt.submission_message_class_id
+      WHERE
+        os.occurrence_submission_id = ${submissionId}
+      ORDER BY sm.submission_message_id;
+    `;
+
+    const response = await this.connection.sql<IOccurrenceSubmissionMessagesResponse>(sqlStatement);
+
+    if (!response?.rows) {
```

Comment:
does this error? ![Array] will never be false. fix: `!response.rows.length`

Reply:
> ![Array] will never be false.

True, but `response.rows` is not guarenteed to be an array; The intent here to branch if `response` has `{ rows: undefined }` which would happen if the query fails. The reason we don't branch on `!response.rows.length` is because `[]` is an expected response